### PR TITLE
Add api calls for wishlist and cart 

### PR DESCRIPTION
**Pull request by SanketDhabarde**
### Changes made
- Add and remove from wishlist using Mockbee
- Add and remove from the cart using Mockbee
- Restrict cart and wishlist route if a user is not logged in
- Add user profile to logout

### User profile
Screenshot (188)

### CODE REVIEW

**Review by maitrakhatri — APPROVED** [review]
Code looks clean and the logic is very easy to understand.

File: src/Components/CartCard/CartCard.jsx [hunk 1]
```diff
@@ -29,6 +31,22 @@ function CartCard({ product }) {
     }
   };
 
+  const handleRemoveFromCart = async (_id) => {
+    if (token) {
+      try {
+        const res = await axios.delete(`api/user/wishlist/${_id}`, {
+          headers: { authorization: token },
+        });
+        console.log(res);
```
Comment (rohanmathur91) [inline src/Components/CartCard/CartCard.jsx]: remove logs

File: src/Components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -22,6 +23,62 @@ function ProductCard({ product }) {
...
+    if (token) {
+      try {
+        const res = await axios.post(
+          `api/user/cart`,
+          { product },
+          {
+            headers: { authorization: token },
+          }
+        );
+        console.log(res);
```
Comment (rohanmathur91) [inline src/Components/ProductCard/ProductCard.jsx]: same as above

**Review by rohanmathur91 — APPROVED** [review]
- File structuring is good
 - Good use of utility functions 
 - Code is clean and easy to understand

**Pull request merged**